### PR TITLE
Add VPA and resources configuration.

### PR DESCRIPTION
**Pull request by weseven**
Towards: https://github.com/giantswarm/roadmap/issues/3625

Adds resource configuration for the cert-exporter deployment and daemonset, adds VPA resource for the deployment (enabled by default).

### CODE REVIEW

**Comment by njuettner** [comment]
What about the daemonset, should we also include it?

**Review by Gacko — APPROVED** [review]
Maybe make VPA optional via a `deployment.autoscaling.enabled` flag, but apart from that: LGTM!

**Comment by weseven** [comment]
> What about the daemonset, should we also include it?

I haven't seen issues on the daemonset side on the resources (it makes sense: it just watches a few files on every node, so it does not suffer as the deployment on big clusters).
So I can make resources configurable for the daemonset, but I won't add the vpa resource for it, wdyt?

**Review by njuettner — APPROVED** [review]

File: helm/cert-exporter/templates/vpa-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{{- if .Values.deployment.autoscaling.enabled }}
```
Comment (Gacko) [inline helm/cert-exporter/templates/vpa-deployment.yaml:1]:
```suggestion
{{- if .Values.deployment.autoscaling.enabled -}}
```

Not 100% sure about that, maybe you want to `helm template` to check, but without this you would get an empty line.

**Review by Gacko — APPROVED** [review]

File: helm/cert-exporter/templates/vpa-deployment.yaml (hunk 1 above)
Comment (Gacko) [reply helm/cert-exporter/templates/vpa-deployment.yaml:1]: Ah, I think this only makes a difference if you have two conditions in the beginning. This is why I told "not sure". I saw Helm rendering an empty line once due to ✨ _circumstances_ ✨.

**Pull request merged**